pkg/master/master.go

```diff
@@ -401,19 +401,10 @@ func setDefaults(c *Config) {
 	if c.CacheTimeout == 0 {
 		c.CacheTimeout = 5 * time.Second
 	}
-	for c.PublicAddress == nil || c.PublicAddress.IsUnspecified() || c.PublicAddress.IsLoopback() {
-		// TODO: This should be done in the caller and just require a
-		// valid value to be passed in.
-		hostIP, err := util.ChooseHostInterface()
-		if err != nil {
-			glog.Fatalf("Unable to find suitable network address.error='%v' . "+
-				"Will try again in 5 seconds. Set the public address directly to avoid this wait.", err)
-			time.Sleep(5 * time.Second)
-			continue
-		}
-		c.PublicAddress = hostIP
-		glog.Infof("Will report %v as public IP address.", c.PublicAddress)
+	if c.PublicAddress == nil || c.PublicAddress.IsUnspecified() || c.PublicAddress.IsLoopback() {
+		glog.Errorf("Invalid public address for master to serve on: %v", c.PublicAddress)
```

Comment:
Yeah, but nobody checks logs.  Seriously.  It's not a big deal, I guess.

On Mon, Nov 23, 2015 at 9:22 PM, Harry Zhang <notifications@github.com>
wrote:

> In pkg/master/master.go
> <https://github.com/kubernetes/kubernetes/pull/17304#discussion_r45697882>
> :
>
> > @@ -401,19 +401,10 @@ func setDefaults(c *Config) {
> >  	if c.CacheTimeout == 0 {
> >  		c.CacheTimeout = 5 * time.Second
> >  	}
> > -	for c.PublicAddress == nil || c.PublicAddress.IsUnspecified() || c.PublicAddress.IsLoopback() {
> > -		// TODO: This should be done in the caller and just require a
> > -		// valid value to be passed in.
> > -		hostIP, err := util.ChooseHostInterface()
> > -		if err != nil {
> > -			glog.Fatalf("Unable to find suitable network address.error='%v' . "+
> > -				"Will try again in 5 seconds. Set the public address directly to avoid this wait.", err)
> > -			time.Sleep(5 * time.Second)
> > -			continue
> > -		}
> > -		c.PublicAddress = hostIP
> > -		glog.Infof("Will report %v as public IP address.", c.PublicAddress)
> > +	if c.PublicAddress == nil || c.PublicAddress.IsUnspecified() || c.PublicAddress.IsLoopback() {
> > +		glog.Errorf("Invalid public address for master to serve on: %v", c.PublicAddress)
>
> api-server already checked this.
>
> But I add this line because I'm afraid when other people add a new caller,
> they may forget they should provide a valid value.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/17304/files#r45697882>.
>